### DISCUSSION TITLE
how we can expose stored procedure of MySQL database in rest API using loopback4

### DISCUSSION THREAD

**Discussion by kanusoni** [loopbackio/loopback-next#7181] [Q&A]
Hello,
is there any library or any code available to get the stored procedure of MySQL database into Rest API using loopback4.

**Comment by dougal83** [comment]
Please see https://github.com/strongloop/loopback-next/issues/5101#issuecomment-613189957

**Comment by deepakrkris** [comment]
@kanusoni please take a look at #3459

constructor(
    @repository(MyRepository)
    public myRepo: MyRepository,
  ) {
    this.user_Profile = userProfile;
  }
 async testData(
  ): Promise<any> {
    return await this.myRepo.dataSource.execute(`EXEC Get_Customer(?)`, ['CUSTID01']);
  }
and

https://loopback.io/doc/en/lb4/apidocs.repository.connector.execute.html

**Comment by kanusoni** [comment]
@deepakrkris @dhmlau That code you mentioned in #3459 is working fine.
But what i want is to get the list of stored procedure and execute the stored procedure of MySQL database using an API in loopback4 explorer.
Thanks.

**Comment by Arathy-sivan** [comment]
What is the use of " --save " in mysql loopback4 connector command??
( npm install --save loopback-connector-mysql )

**Comment by achrinza** [comment]
@Arathy-sivan This is a question with regards to NPM and not specific to LoopBack 4.

Please see https://stackoverflow.com/questions/19578796/what-is-the-save-option-for-npm-install
